pkg/apis/resource/types.go

```diff
@@ -71,6 +71,15 @@ type ResourceSlice struct {
 	Spec ResourceSliceSpec
 }
 
+const (
+	// ResourceSliceSelectorNodeName can be used in a [metav1.ListOptions]
+	// field selector to filter based on [ResourceSliceSpec.NodeName].
+	ResourceSliceSelectorNodeName = "nodeName"
```

Comment:
field selectors, As far as I understand, are not supposed to be magic constants - they are paths into the type-in-question, so `spec.nodeName` and `spec.driver` ?

Reply:
Either way is fine with me.

Pod uses `nodeName` as field selector although the field is `spec.nodeName`, but that's probably a historic accident that we shouldn't repeat.

Reply:
Where do you see that?  I see things like:

```
pkg/registry/core/pod/strategy_test.go:                 fieldSelector: fields.ParseSelectorOrDie("spec.nodeName=nodeA"),
staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go:                      fieldSelector: fields.ParseSelectorOrDie("spec.nodeName=t2-bar1"),
cmd/kubeadm/app/util/config/cluster.go:                 FieldSelector: fmt.Sprintf("spec.nodeName=%s", nodeName),
staging/src/k8s.io/kubectl/pkg/drain/drain.go:          FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": nodeName}).String(),
test/e2e/storage/non_graceful_node_shutdown.go:                         FieldSelector: fields.OneTermNotEqualSelector("spec.nodeName", oldNodeName).String(),
```

Reply:
You are right. Sorry, got confused (zombie mode...). It's still one my TODO list for fixing.

Reply:
Fixed.

I am keeping the consts in the API, for two reasons:
- They document that these fields can be used as field selector. That is not at all clear in other Kubernetes APIs.
- I don't like that Go code using such APIs is using literal strings. This isn't wrong because these strings are part of the API and cannot change, but it's not obvious that these strings are correct (typos!) and where they come from.

Reply:
They are part of the API, I agree with that.  If there ever was a v2 which moved or changed this, the const would be versioned with it.  That's fine.  It's effectively an index-name or WHERE clause.